src/vmm/src/vstate/vcpu/aarch64.rs

```diff
@@ -182,21 +183,50 @@ impl KvmVcpu {
 #[derive(Debug, Default, Clone, Versionize)]
 pub struct VcpuState {
     pub mp_state: kvm_bindings::kvm_mp_state,
-    pub regs: Vec<Aarch64Register>,
+    #[version(end = 2, default_fn = "default_old_regs")]
+    old_regs: Vec<Aarch64RegisterOld>,
+    #[version(start = 2, de_fn = "de_regs", ser_fn = "ser_regs")]
+    pub regs: Aarch64RegisterVec,
     // We will be using the mpidr for passing it to the VmState.
     // The VmState will give this away for saving restoring the icc and redistributor
     // registers.
     pub mpidr: u64,
 }
 
+impl VcpuState {
+    fn default_old_regs(_: u16) -> Vec<Aarch64RegisterOld> {
+        Vec::default()
+    }
+
+    fn de_regs(&mut self, _source_version: u16) -> VersionizeResult<()> {
+        let mut regs = Aarch64RegisterVec::default();
+        for reg in self.old_regs.iter() {
+            regs.push(reg.as_reg_ref());
+        }
+        self.regs = regs;
+        Ok(())
+    }
+
+    fn ser_regs(&mut self, _target_versio: u16) -> VersionizeResult<()> {
+        self.old_regs = self
+            .regs
+            .iter()
+            .map(Aarch64RegisterOld::from_reg_ref)
```

Comment:
what's going to happen here if we feed registers bigger than 128 bit?

Reply:
Changed `from_reg_ref` to return a error if conversion is not possible.